src/main/java/strategy/operation/OperationHandler.java

```diff
@@ -0,0 +1,8 @@
+package strategy.operation;
+
+import java.util.Map;
+import model.FruitTransaction;
+
+public interface OperationHandler {
+    void execute(Map<String, Integer> fruitBalance, FruitTransaction transaction);
```

Comment:
do not pass the map of fruit balance here, it has to be available by calling Dao methods inside execute method